src/Symfony/Component/String/AbstractString.php

```diff
@@ -0,0 +1,737 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\String;
+
+use Symfony\Component\String\Exception\ExceptionInterface;
+use Symfony\Component\String\Exception\InvalidArgumentException;
+
+/**
+ * Represents a string.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ * @author Hugo Hamon <hugohamon@neuf.fr>
+ *
+ * @throws ExceptionInterface
+ */
+abstract class AbstractString implements \JsonSerializable
+{
+    protected $string = '';
+    protected $ignoreCase = false;
+
+    abstract public function __construct(string $string = '');
+
+    /**
+     * @param string|array $needle
```

Comment:
From what i can tell all the `array` type hints mean an array of strings. How about typehinting those to `string[]`? or `array<string>`. That should help users and any analysis tools that they may use.

Reply:
updated thanks